File: python/rna-transcription/rna_transcription.py
def to_rna(dna_strand):
    complements = {'G': 'C', 'C': 'G', 'T': 'A', 'A': 'U'}

    if dna_strand == '':
        return ''
    elif len(dna_strand) == 1:
        return complements[dna_strand]

    rna = ''
    for d in dna_strand:
        if complements[d]:
            rna += complements[d]
    return rna

File: go/rna-transcription/rna_transcription.go
package strand

var complements = map[rune]string{
	'G': "C",
	'C': "G",
	'T': "A",
	'A': "U",
}

func ToRNA(dna string) string {
	var newStrand string
	for _, strand := range dna {
		c := complements[strand]
		newStrand += c
	}
	return newStrand
}
